runtime/src/main/java/io/quarkus/reactivemessaging/http/runtime/HttpSink.java

```diff
@@ -103,17 +105,27 @@ private <T> Buffer serialize(T payload) {
     }
 
     private Uni<Void> invoke(HttpRequest<?> request, Buffer buffer) {
+        log.debugf("Invoking request: ", toString(request, buffer));
         return request
                 .sendBuffer(buffer)
                 .onItem().transform(resp -> {
                     if (resp.statusCode() >= 200 && resp.statusCode() < 300) {
                         return null;
                     } else {
-                        throw new RuntimeException("HTTP request returned an invalid status: " + resp.statusCode());
+                        throw new VertxException(
```

Comment:
Not sure if we can throw an exception that's not from our domain. Why can't we keep the RuntimeException instead? What does VertxException brings to the table?

Reply:
I was thinking about this and I have the same reservations, but came up with the following rational, 
I think a domain exception (although generic) gives a clue to guys watching the exception in the logs about the technology we are using. 
One real usage case, an user add http apache log level config debug hoping to see the request and he obviously obtain nothing in return because we are using vertx (which does not have any trace, hence the need to do this PR)
He believed we were using Apache because the exception we are throwing is Runtime, so it was a reasonable bet. 
Actually I have doubt we should throw an exception at all but I did not want to alter that. Assuming we have to throw an exception, I beilieve it should actually be domain specific. Since this a client scenario, it should not cause any classpath issue.

Reply:
Customer that catching Runtime wont be affected, becuase Verx is a runtime one. 
The only problem is that a user catches Vertx and after that we switch to Http Apache, his code will stop compiing, so in that sense, it is a better to throw a JDK one (and keep that one unchanged)
Our choice ;)